sdk/confidentialledger/Azure.Storage.ConfidentialLedger/src/Generated/ConfidentialLedgerClient.cs

```diff
@@ -48,7 +48,7 @@ public ConfidentialLedgerClient(Uri ledgerUri, TokenCredential credential, Confi
             _clientDiagnostics = new ClientDiagnostics(options);
             _tokenCredential = credential;
             var authPolicy = new BearerTokenAuthenticationPolicy(_tokenCredential, AuthorizationScopes);
-            Pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { authPolicy, new LowLevelCallbackPolicy() });
+            Pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() }, new HttpPipelinePolicy[] { authPolicy }, new ResponseClassifier());
```

Comment:
Why would auth policy need to be per-retry? That said, this seems like a very targeted change given only a few client libraries are affected, so I'm signing off to unblock assuming this is what is desired. But I did want to call it out as odd.

Reply:
That's the planned "default" for LLC, we setup the LowLevelCallbackPolicy to provide per call notifications, and the auth like that.

Hey @ellismg @KrzysztofCwalina - Confirmation that this is desired?

Reply:
Most auth policies I've seen attach a bearer token to the header. How many times do you need to do that? Seems only once, given that headers are persisted across retries (same Request object).

/cc @pakrym

Reply:
Note that the overload we were calling already treated all the policies as per-retry, not per call: (https://github.com/Azure/azure-sdk-for-net/blob/2e13fe427dcf4c9032aa1ae12e5de4f9b8acbf88/sdk/core/Azure.Core/src/Pipeline/HttpPipelineBuilder.cs#L22-L35)

As to why the authentication policy is per retry vs per call?  I assume it has something to do with the fact that the auth policy may need to sign the request in some way and that singing may only be valid per retry (for example, you might include a timestamp in your signature and that timestamp may need to be close to the time the HTTP request was invoked, not the time the whole operation started (you could imagine for AAD that the per-retry semantics are also helpful as it may force the token to be refreshed if it has expired between retrys).

Put another way, I don't think this is changing anything about how the auth policy is handled in general, from a quick `git grep` it looks like we've always been on this plan, even for non LLC clients.

Reply:
The auth policy here, though, is a bearer token policy and doesn't sign the request. If the auth policy varies and how it's added is more generalized I could see that, but is that the case? As long as the bearer token was properly cached in time (should be), I suppose there's no harm apart from some negligable synchronized cache hits (compared to network calls), but we should double check this is indeed what's required/desired.

Reply:
> (you could imagine for AAD that the per-retry semantics are also helpful as it may force the token to be refreshed if it has expired between retrys).

This is my understanding of why auth policies should always be pre-retry. If the client is using the BearerTokenAuthenticationPolicy, the header is cached anyway, assuming it is not nearing expiration.

Reply:
I don't think this discussion blocks landing this PR, so i'm going to land it.

If we want to change how LLC does Auth or discuss it more let me know and i'll setup a meeting.